sdk/js-query/src/query/utils.ts

```diff
@@ -0,0 +1,31 @@
+import * as elliptic from "elliptic";
+
+export function isHex(s: string) {
+  // TODO: even length?
+  return /^(0x)?[0-9a-fA-F]+$/.test(s);
+}
+
+export function hexToUint8Array(s: string): Uint8Array {
+  if (!isHex(s)) {
+    throw new Error(`${s} is not hex`);
+  }
+  return new Uint8Array(
+    Buffer.from(s.startsWith("0x") ? s.slice(2) : s, "hex")
```

Comment:
May need to append a leading '0' if the length is odd:
```> let b = Buffer.from("fff", "hex")
undefined
> b
<Buffer ff>
> let bb = Buffer.from("0fff", "hex")
undefined
> bb
<Buffer 0f ff>```

Reply:
We assert that the length is even now in `isValidHexString`